internal/api/http/module.go

```diff
@@ -0,0 +1,30 @@
+package http
+
+import (
+	"context"
+
+	"GoEdu/internal/api/http/controller"
+	"GoEdu/internal/api/http/server"
+	"GoEdu/internal/pkg/module"
+)
+
+type Module struct{}
+
+func (m *Module) Init(ctx context.Context, dep module.Dependencies) error {
+	return Root(ctx, dep)
+}
+
+func Root(ctx context.Context, dep module.Dependencies) error {
+	mux := dep.HTTP()
+	tp := dep.TraceProvider()
+	log := dep.Logger()
+
+	impl := controller.New(tp.Tracer("controller"))
+	_ = server.HandlerWithOptions(impl, server.StdHTTPServerOptions{
+		BaseRouter: mux,
+		Middlewares: []server.MiddlewareFunc{
+			controller.NewLoggerHTTPMiddleware(log.With("module", "http-api")),
+		},
+	})
+	return nil
+}
```

Comment:
How it works?

Reply:
According to the [docs](https://github.com/deepmap/oapi-codegen?tab=readme-ov-file#go-122-nethttp) the code generation tool generates go1.22+ http interface for each of oapi routes. Also it generates types of oapi json models. That interface we have to implement. The implementation lays down in `controller` package. After we provided the implementation, we can register our generated routes to provided `http.ServeMux` router. Also we can add some middlewares too.